### ISSUE TITLE
flows: ConjugateTransformFilter — ensemble Kalman analysis in a Gaussianised latent space

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#144]
## Problem / Request

The ensemble Kalman filter's Gaussian update is not wrong in itself — it is
applied in the wrong **coordinates**. If the prior is the pushforward of a
Gaussian under a bijection `Γ`, and the likelihood is Gaussian in the latent
space, then the Kalman update performed in latent space and mapped back is
**exactly Bayes**.

Add `ConjugateTransformFilter`: warp the ensemble, call `gaussx.enkf_analysis`,
warp back. This is the smallest issue in the set and the largest accuracy win.

## User Story

> As a user assimilating observations of a strictly-positive or bounded quantity,
> I want the ensemble update performed in a Gaussianised space, so that my
> posterior mean is not biased by ~10% in a way that no amount of extra ensemble
> members will fix.

## Motivation

Verified against 3000×3000 grid quadrature on the reference paper's own
log/logit example (`nkf_verify.py` §6):

| quantity | latent-space update | grid reference | error |
|---|---|---|---|
| posterior mean | `[-0.624444, -0.643518]` | identical | **2.2e-16** |
| posterior covariance | `[[0.046154, 0.019231], [0.019231, 0.303846]]` | identical | **1.6e-16** |

And the cost of the wrong coordinates, at 2,000,000 ensemble members so sampling
noise is not the story:

| method | `E[x₁]` | `E[x₂]` |
|---|---|---|
| exact posterior | 0.548062 | 0.353937 |
| **EnKF in physical space** | **0.607391 (+10.8%)** | **0.388017 (+9.6%)** |
| **ECTF (same ensemble, latent update)** | **0.547980** | **0.353964** |

A bias that does not shrink with ensemble size, removed by conjugating the update
with a bijection.

This also gives `fit_rbig` a second job it is unusually well suited to: RBIG
produces a closed-form Gaussianising map from the prior ensemble with no training
loop, which is exactly the `Γ` the method needs and exactly what the source paper
hand-specifies per experiment.

## Proposed API

```python
class ConjugateTransformFilter(eqx.Module):
    r"""Ensemble Kalman analysis performed in a Gaussianised latent space.

    Implements the Ensemble Conjugate Transform Filter of Chipilski (2025).
    Given a bijection $\Gamma$ that maps observations and states into a
    space where the prior is Gaussian, the analysis step is

    $$
    X^a = \Gamma\!\left(\mathrm{EnKF}\!\left(\Gamma^{-1}(X^f),\;
          \Gamma^{-1}(Y^f),\; \Gamma^{-1}(y)\right)\right).
    $$

    When the prior is exactly the pushforward of a Gaussian under $\Gamma$
    and the likelihood is Gaussian in the latent space, this update is
    **exact Bayes**, not an approximation -- the ensemble Kalman filter's
    Gaussian assumption is a statement about coordinates, not about the
    algorithm.

    Attributes:
        warp: Bijection mapping latent (Gaussian) values to physical values.
            FlowJax convention: ``transform`` is latent -> physical,
            ``inverse`` is physical -> latent.
        obs_warp: Optional separate bijection for observation space. Defaults
            to ``warp`` when state and observation live in the same space.
    """

    warp: AbstractBijection
    obs_warp: AbstractBijection | None = None

    def analysis(
        self,
        particles: Float[Array, "J N"],
        obs_particles: Float[Array, "J M"],
        observation: Float[Array, " M"],
        obs_noise: lx.AbstractLinearOperator,
        *,
        key: PRNGKeyArray,
        localization: Float[Array, "N M"] | None = None,
    ) -> Float[Array, "J N"]:
        """Analysis step. Returns the posterior ensemble in physical space."""


def rbig_conjugate_filter(
    particles: Float[Array, "J N"],
    *,
    key: PRNGKeyArray,
    n_layers: int = 10,
) -> ConjugateTransformFilter:
    """Fit a Gaussianising warp to a prior ensemble with RBIG.

    Convenience constructor: RBIG gives a closed-form $\\Gamma$ from the
    ensemble with no gradient-based training, which is what the conjugate
    transform needs and what the reference implementation specifies by hand
    for each experiment.
    """
```

## Design Snapshot

The whole method:

```python
def analysis(self, particles, obs_particles, observation, obs_noise, *, key,
             localization=None):
    obs_warp = self.obs_warp if self.obs_warp is not None else self.warp

    z   = jax.vmap(self.warp.inverse)(particles)          # (J, N) latent state
    g   = jax.vmap(obs_warp.inverse)(obs_particles)       # (J, M) latent obs
    y_g = obs_warp.inverse(observation)                   # (M,)   latent obs value

    z_a = gaussx.enkf_analysis(
        z, g, y_g, obs_noise, perturbed_obs=..., localization=localization,
    )
    return jax.vmap(self.warp.transform)(z_a)             # back to physical space
```

Reference implementation — note that the *newer* of the two variants in the source
repo takes the perturbed observation ensemble directly rather than an explicit `R`,
which is why `enkf_analysis` exposes `perturbed_obs`:

```python
# hchipilski/CTF, ECTF.py:36-49  (ECTF_scalarOb)
HXfG   = H.dot(XfG)
SHt    = calc_norm_pert_ens(XfG).dot(calc_norm_pert_ens(HXfG).T)
HSHt_R = calc_norm_pert_ens(YfG).dot(calc_norm_pert_ens(YfG).T)
K      = SHt.dot(inv_2by2(HSHt_R))
XaG    = XfG + K.dot(yG_arr - YfG)
Xa     = np.array([np.exp(XaG[0, :]), logistic(XaG[1, :])])   # <- the warp, hardcoded
```

The last line is the entire idea, written as a hardcoded `exp` / `logistic` pair.
Replacing it with an `AbstractBijection` is the port.

## Mathematical Notes

```text
Setup. Prior p(x) is the pushforward of N(μ, Σ) under a bijection Γ:

        x = Γ(ζ),        ζ ~ N(μ, Σ)

Likelihood is Gaussian in latent coordinates:

        Γ⁻¹_obs(y) | ζ  ~  N(Hζ, R)

Then the posterior over ζ is Gaussian by conjugacy and the Kalman update is
EXACT. Pushing forward through Γ gives the exact posterior over x. Verified to
2.2e-16 (mean) and 1.6e-16 (covariance) against grid quadrature.

Worked example (the reference paper's, and the test case):

        prior:  (log x₁, logit x₂) ~ N(μ, Σ)
        obs:    multiplicative lognormal noise on x₁, i.e. log y = log x₁ + ε

  latent posterior mean = [−0.624444, −0.643518]
  latent posterior cov  = [[0.046154, 0.019231], [0.019231, 0.303846]]

CAVEAT to state in the docstring. Exactness requires the likelihood to be
Gaussian in the SAME latent coordinates that Gaussianise the prior. When it is
not, this is still a strictly better approximation than the physical-space
update, but it is an approximation. The library cannot check this condition;
say so rather than implying a guarantee.
```

## References & Existing Code

- Design doc: `kalman_survae_report.md` §1.3, §2.4, §4-F
- Paper: Chipilski, H. G. (2025), *Exact Nonlinear State Estimation*, J. Atmos. Sci. 82(4) 809–827, [doi:10.1175/JAS-D-24-0171.1](https://doi.org/10.1175/JAS-D-24-0171.1)
- Reference impl: [`hchipilski/CTF`](https://github.com/hchipilski/CTF) `ECTF.py` (the method), `EnKF.py` (the baseline), `distributions.py:282` (`bayesian_update_2D`, the analytic reference the tests should reuse)
- Verification script: `nkf_verify.py` §6
- RBIG warm start: `src/gauss_flows/_src/init/rbig.py`, `src/gauss_flows/_src/flows/rbig.py`
- Consumed upstream symbol: `gaussx.enkf_analysis`

## Implementation Steps

- [ ] Add `ConjugateTransformFilter` and `rbig_conjugate_filter` in `src/gauss_flows/_src/flows/conjugate_filter.py`
- [ ] Export both from `src/gauss_flows/_src/flows/__init__.py` and `src/gauss_flows/__init__.py`
- [ ] Add `::: gauss_flows.ConjugateTransformFilter` and `::: gauss_flows.rbig_conjugate_filter` to `docs/api/flows.md`
- [ ] Validate that `warp` is genuinely invertible on the ensemble range — a `NumericalInverse`-backed warp will be slow inside `vmap`; warn or document
- [ ] State the "likelihood must be Gaussian in the same latent coordinates" caveat prominently in the class docstring, not just the module docstring

## Definition of Done

- [ ] Code lands at the intended path
- [ ] Public API exported via `src/gauss_flows/__init__.py`
- [ ] Tests pass: `make test`
- [ ] Lint + typecheck pass: `make lint && make typecheck`
- [ ] Docstrings (Google-style) on all public symbols

## Testing

New file `tests/test_flows_conjugate_filter.py`.

- [ ] **Exactness test:** on the log/logit example, the latent posterior mean and covariance match grid quadrature — **target ≤ 1e-12** (measured 2.2e-16 / 1.6e-16). Build the grid reference in the test rather than hardcoding, so it stays a real check
- [ ] **Improvement test:** with the same 200k-member ensemble and the same PRNG key, `ConjugateTransformFilter` posterior mean error is at least 50× smaller than plain `gaussx.enkf_analysis`. Measured at 2M members: ECTF `[0.547980, 0.353964]` vs EnKF `[0.607391, 0.388017]` against exact `[0.548062, 0.353937]`
- [ ] Identity test: an `Identity` warp reproduces `gaussx.enkf_analysis` bit-for-bit
- [ ] Convergence test: ECTF posterior-mean error shrinks as `J → ∞`, while the physical-space EnKF error plateaus. Sweep `J ∈ {2e3, 2e4, 2e5}` and assert the plateau — this is the property that makes the method worth having, and a mean-only test at a single `J` does not show it
- [ ] Round-trip: `warp.transform(warp.inverse(x)) == x` on the ensemble range, ≤ 1e-10
- [ ] Separate `obs_warp` test: state and observation spaces with different warps produce the correct latent innovation
- [ ] `rbig_conjugate_filter` test: a fitted RBIG warp on a lognormal ensemble reduces posterior-mean bias relative to no warp (looser tolerance — RBIG is fitted, not exact)
- [ ] Shape test: `(J, N)` in, `(J, N)` out, `N ≠ M`
- [ ] JIT test: `eqx.filter_jit(analysis)` returns finite values
- [ ] Use `jr.key(...)`, not `jr.PRNGKey(...)`

## Documentation

- [ ] `docs/api/flows.md` — section framing the method as "the EnKF's Gaussian assumption is about coordinates", with the bias table
- [ ] Notebook `docs/notebooks/` — reproduce Fig. 5 of Chipilski (2025): EnKF vs ECTF on the lognormal/logit-normal prior, against the analytic posterior
- [ ] Docstrings (covered by Definition of Done)

## Out of scope

Do **not** port QCEF (`hchipilski/CTF/QCEF.py`). It is the baseline the source
paper argues against: it CDF-transforms the observed variable but then propagates
the increment to unobserved variables through a linear regression coefficient
`cov(X,Y)/var(Y)` computed in **physical** space (`QCEF.py:52-66`), which is
exactly the step this method removes. It is in the reference repo to lose the
comparison.

## Relationships

- Parent (theme epic): #
- Blocked by: # (gaussx: enkf_analysis), # (gauss_flows: LGSSM base distributions — for the shared `gaussx` dependency)
- Blocks: #
- Related: # (gauss_flows: NKF composition)
